bin/get_data.sh

```diff
@@ -0,0 +1,127 @@
+init_date=$(date +%Y-%m-%dT%H:%M:%S)
+if [[ "$init_date" > "$DATE" ]] 
+then 
+    init_date="${DATE}T${TIME}"
+fi
+export time_units="hours since ${init_date}"
+
+# Download ml, sfc, pv and pt files
+echo "Downloading files, this might take a long time!"
+if [ ! -f grib/${BASE}.ml.grib ]; then
+    echo test
+    python bin/download_an_all.py $DATE $TIME ml &
+fi
+if [ ! -f grib/${BASE}.sfc.grib ]; then
+    python bin/download_an_all.py $DATE $TIME sfc &
+fi
+if [ ! -f grib/${BASE}.pv.grib ]; then
+    python bin/download_an_all.py $DATE $TIME pv &
+fi
+if [ ! -f grib/${BASE}.tl.grib ]; then
+    python bin/download_an_all.py $DATE $TIME tl &
+fi
+wait
+
+if [ ! -f grib/${BASE}.ml.grib ]; then
+   echo	FATAL `date` Model level file is missing
+   exit
+fi
+if [ ! -f grib/${BASE}.sfc.grib ]; then
+   echo	FATAL `date` Surface file is missing
+   exit
+fi
+if [ ! -f grib/${BASE}.pv.grib ]; then
+   echo	FATAL `date` Potential Vorticity level file is missing
+   exit
+fi
+if [ ! -f grib/${BASE}.tl.grib ]; then
+   echo	FATAL `date` Potential Temperature level file is missing
+   exit
+fi
+
+cat grib/${BASE}.ml.grib grib/${BASE}.sfc.grib > ${GRIB}
+
+# convert grib to netCDF, set init time
+cdo -f nc4c copy grib/${BASE}.tl.grib $tlfile
+ncatted -a units,time,o,c,"${time_units}" $tlfile
+cdo -f nc4c copy grib/${BASE}.pv.grib $pvfile
+ncatted -a units,time,o,c,"${time_units}" $pvfile
+cdo -f nc4c copy ${GRIB} $mlfile
+ncatted -a units,time,o,c,"${time_units}" $mlfile
+
+# Add pressure and geopotential height to model levels file
+bin/add_pressure_gph.sh input=$mlfile pressure_units=Pa gph_units="m^2s^-2"
+
+# Add ancillary information
+python bin/add_ancillary.py $mlfile --pv --theta --tropopause --n2 #--eqlat nan values cause issues for now due to no 180° coverage
+
+# separate sfc from ml variables
+ncks -7 -L 7 -C -O -x -vlev_2,n2,clwc,u,q,t,pressure,zh,cc,w,v,ciwc,pt,pv,mod_pv,o3,d $mlfile $sfcfile
+ncatted -O -a standard_name,msl,o,c,air_pressure_at_sea_level $sfcfile
+ncks -C -O -vtime,lev_2,lon,lat,n2,clwc,u,q,t,pressure,zh,cc,w,v,ciwc,pt,pv,mod_pv,o3,d,hyai,hyam,hybi,hybm,sp,lnsp $mlfile $tmpfile
+mv $tmpfile $mlfile
+
+# interpolate to different grids
+echo "Creating pressure level file..."
+cdo ml2pl,85000,50000,40000,30000,20000,15000,12000,10000,8000,6500,5000,4000,3000,2000,1000,500,100 $mlfile $plfile
+ncatted -O -a standard_name,plev,o,c,atmosphere_pressure_coordinate $plfile
+ncap2 -s "plev/=100;plev@units=\"hPa\"" $plfile $plfile-tmp
+mv $plfile-tmp $plfile
+ncks -7 -L 7 -C -O -x -v lev,sp,lnsp,nhyi,nhym,hyai,hyam,hybi,hybm $plfile $plfile
+
+echo "Creating potential temperature level file..."
+python bin/interpolate_missing_variables.py $mlfile $tlfile pt
+python bin/rename_standard.py $mlfile $tlfile
```

Comment:
Das rename_standard Skript ist seltsam benamt und mir ist nicht ganz klar warum man es braucht.
Am ml-file hast Du dich kaum rumgefummelt, warum sind die Attribute da drin besser?

Reply:
Einige Standardnamen und Units sind in ml festgelegt und in allen anderen nicht.
Statt 50 Zeilen lang alle Standardnamen und Units in allen files zu ersetzen, ersetze ich die 1 mal in ml und benutze das Skript.